cpp/src/cider/util/CiderParquetReader.h

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright(c) 2022-2023 Intel Corporation.
+ * Copyright (c) OmniSci, Inc. and its affiliates.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#ifndef CIDER_PARQUETREADER_H
+#define CIDER_PARQUETREADER_H
+
+#include <arrow/api.h>
+#include <arrow/filesystem/api.h>
+#include <arrow/io/api.h>
+#include <arrow/util/cpu_info.h>
+#include <arrow/util/logging.h>
+#include <parquet/api/reader.h>
+#include <string>
+#include <vector>
+#include "exec/plan/parser/TypeUtils.h"
+#include "util/ArrowArrayBuilder.h"
+#include "util/Logger.h"
+
+namespace CiderParquetReader {
+
+enum FileSystemType { LOCAL, HDFS, S3 };
```

Comment:
use enum class instead and use `kLocal, kHdfs, kS3`
https://google.github.io/styleguide/cppguide.html#Enumerator_Names